hr_expense_adj/models/hr_expense.py

```diff
@@ -84,3 +85,15 @@ def _compute_account_id(self):
                 'expense']
             if account:
                 exp.account_id = account
+
+    @api.constrains('date', 'analytic_account_id')
+    def _validate_analytic_date(self):
+        for rec in self:
+            projects = rec.analytic_account_id.with_context(
+                active_test=False).mapped('project_ids')
+            if projects:
```

Comment:
@TimLai125 With this, there is no validation when project is not found, which could be an issue when there is actually a project but user does not have the right to access (I think it may happen depending on the project visibility).

Reply:
Add sudo with  #134